### PR TITLE
Moved login logic into triggerable method

### PR DESCRIPTION
**Pull request by adam-cowley**
I've moved the code from inside the route into a triggerable handler that can be used across the Multicolour environment.  For example:

```
 server.route({
      method: "POST",
      path: "/login",
      config: {
        auth: false,
      },
      handler: (request, reply) => {
        const args = {
          email: request.payload.email.toString(),
          password: request.payload.password,
          callback: (err, session) => {
            const redirect_to = request.query.redirect || '/';
            const cookieKey = config.cookieKey || DEFAULT_COOKIE_TOKEN;

            if (err) {
              return reply.redirect(`/login?redirect=${encodeUriComponent(redirect_to)}&error=true`);
            }

            // Apply Redirect
            reply.redirect(redirect_to)
              // Set Cookie as per jwt2 docs so we can pick it up in the front end
              .state(cookieKey, session.token)
          }
        }

        generator.trigger('auth_login', args);
      }
    })
  }
```

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "multicolour-hapi-jwt",
-  "version": "0.1.1",
+  "version": "0.1.2",
```
Comment (davemackintosh) [inline package.json]: No need to bump version for a PR, I will bump once merged for publishing.

File: index.js [hunk 2]
```diff
@@ -38,6 +41,62 @@ class Multicolour_Auth_JWT {
       })
   }
 
+  auth(multicolour, identifier, password, callback, identifier_field = 'email') {
```
Comment (davemackintosh) [inline index.js]: No need to pass `multicolour` in here, this isn't a static method so you have access to `this.request("host")` which is the hosting instance of multicolour.

Default argument values aren't in LTS and all multicolour modules should have LTS support. Please move this to a conditional  in the function body.

File: index.js [hunk 3]
```diff
@@ -38,6 +41,62 @@ class Multicolour_Auth_JWT {
...
+    const jwt = require("jsonwebtoken")
+
+    const config = multicolour.get("config").get("auth")
+
+    models
+      .multicolour_user.findOne({
+        [identifier_field]: identifier,
+        requires_password: false
+      })
+      .catch(err => {
```
Comment (davemackintosh) [inline index.js]: You don't need to create a function to call the callback unless you specifically don't want a return value post `.catch`. Just use `.catch(callback)`

File: index.js [hunk 4]
```diff
@@ -38,6 +41,62 @@ class Multicolour_Auth_JWT {
...
+              provider: "jwt",
+              user: user.id,
+              token
+            }, (err, session) => {
+              // Check for errors.
+              if (err) {
+                callback(err);
+              }
+              else {
+                multicolour.trigger('auth_session_created', session);
```
Comment (davemackintosh) [inline index.js]: Please don't mix single and double quotations. Always double quotations.

File: index.js [hunk 5]
```diff
@@ -62,15 +121,27 @@ class Multicolour_Auth_JWT {
       .request("header_validator")
         .set("authorization", joi.string().required())
 
+    generator.on('auth_login', args => {
```
Comment (davemackintosh) [inline index.js]: double quotes here.

File: index.js [hunk 6]
```diff
@@ -81,66 +152,29 @@ class Multicolour_Auth_JWT {
 
     // Headers for the session endpoints.
     const headers = host.request("header_validator").get()
-    delete headers.authorization
+    delete headers.authorization;
```
Comment (davemackintosh) [inline index.js]: No semi-colons https://github.com/Multicolour/multicolour-auth-jwt/blob/master/.eslintrc

File: index.js (hunk 2 above)
Comment (adam-cowley) [inline index.js]: I think this came from the original plugin, I will fix.

File: index.js (hunk 3 above)
Comment (adam-cowley) [inline index.js]: This needs moving to the bottom of the function also. Will fix

**Comment by davemackintosh** [comment]
Could you merge the upstream master into your master? Will solve the conflict on Github @adam-cowley

**Pull request merged**